### PR TITLE
feat(sdk): add CD to kotlin SDK

### PR DESCRIPTION
**Pull request by anonvt**

### CODE REVIEW

File: .github/workflows/publish-to-maven.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          sudo udevadm trigger --name-match=kvm
+
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      - name: Make Gradle executable
+        run: chmod +x ./gradlew
```
Comment (jbergstroem) [inline .github/workflows/publish-to-maven.yaml]: Why not check in this file as executable?

Comment (anonvt) [reply .github/workflows/publish-to-maven.yaml]: It's already checked as, I'll remove this.
I'll also remove the KVM, it speeds up testing for connected android tests, but it's not relevant for build-release.

File: .github/workflows/publish-to-maven.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+on:
+  release:
+    types: [published]
+
+jobs:
+  publishToMaven:
+    environment:
+      name: "MavenCentral"
+
+    runs-on: ubuntu-latest
```
Comment (jbergstroem) [inline .github/workflows/publish-to-maven.yaml]: Be implicit:
```suggestion
    runs-on: ubuntu-24.04
```

File: .github/workflows/publish-to-maven.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    timeout-minutes: 30
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
```
Comment (jbergstroem) [inline .github/workflows/publish-to-maven.yaml]: Is this required? https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#java

**Review by jbergstroem — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-to-maven.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Upload library
+        run: ./gradlew :TopsortAnalytics:publishAndReleaseToMavenCentral
```
Comment (jbergstroem) [inline .github/workflows/publish-to-maven.yaml:22]: Is it possible to split up build and upload? I'd rather not mix the two.

File: .github/workflows/publish-to-maven.yaml (hunk 3 above)
Comment (anonvt) [reply .github/workflows/publish-to-maven.yaml]: Putting the ubuntu version explicitly, java 17 is implict, so I'll remove this

File: .github/workflows/publish-to-maven.yaml (hunk 4 above)
Comment (anonvt) [reply .github/workflows/publish-to-maven.yaml:22]: Split off :compileReleaseKotlin which is the build step that takes the longest (and it's where problems usually show up)

**Review by jbergstroem — APPROVED** [review]

**Pull request merged**